docs/wiki/framework/attach-framework.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: wiki
+title: Attach Framework
+description: Explains how to add items to ACE's Attach system.
+group: framework
+order: 0
+parent: wiki
+mod: ace
+version:
+  major: 3
+  minor: 17
+  patch: 0
+---
+
+## 1. Config Values
+### 1.1 Make item attachable
+
+An item can be added to ACEs "Attach to Vehicle" and "Attach to Self" system by adding the following config value to the ``CfgWeapons`` or ``CfgMagazines`` configs:
+```cpp
+class CfgWeapons {
+    class attach_item: CBA_MiscItem {
+        ACE_attachable = "new_attachable_item_classname";
+    };
+};
+```
+
+The classname in ``ACE_attachable`` has to match a config value in ``CfgVehicle`` which to be able to spawn the physical item in the world. Example:
+```cpp
+class CfgVehicles {
+    class ThingX;
+	// The object that gets created and attached in ACE
+    class new_attachable_item_classname: ThingX {
+        scope = HIDDEN;
+        displayName = "New ACE attachable item";
+        model = QPATHTOF(data\model_file.p3d);
+		vehicleClass = "";
+    };
+};
+```
```

Comment:
```suggestion
```cpp
class CfgWeapons {
    class attach_item: CBA_MiscItem {
        ACE_attachable = "new_attachable_item_classname";
    };
};

class CfgVehicles {
    class ThingX;
    class new_attachable_item_classname: ThingX {
        scope = 1; // Should be 1 (private) or 2 (public), scope 0 will cause errors on object creation
        displayName = "New ACE attachable item";
        model = "\path\to\my\model.p3d";
		vehicleClass = "";
    };
};
```
```